### DISCUSSION TITLE
Get all table names via supabase-js client.

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [supabase/supabase#8948] [Questions] [answered]
Hi there,

I wan't to dynamically get all the table names from my db using the JS client side api is this possible or am I trying to do something a little bit strange.

**Comment by Edditoria** [comment, marked as answer]
Possible. Try to create a "Database Functions" and, for example, call the function using Javascript.

# Create Database Functions

You can use **SQL Editor** in Supabase website. Add a SQL snippet:

```sql
create or replace function get_all_table_name()
returns setof record
as $$
  -- select * from information_schema.tables
  select table_name from information_schema.tables
  where table_schema = 'public'; -- and table_type = 'BASE TABLE';
$$ language sql;
```

Click "RUN" button in the page. You'll see "Success. No rows returned", but it is normal.
Then, add another SQL snippet:

```sql
select get_all_table_name()
```

Click "RUN" and you'll see a table listing the table name :)

# Consume Using Javascript

```js
const { data, error } = await supabase.rpc('get_all_table_name');
```

Reference: https://supabase.com/docs/guides/database/functions
